### ISSUE TITLE
Stronger/Tighter grouping of related versions in workloads listing pages

### ISSUE THREAD

**Issue by mtho11** [kiali/kiali#2806]
**Is your feature request related to a problem? Please describe.**
I would like to see a stronger/tighter more visually related grouping of workloads there is nothing to show that that they are related (by version). 

![Kiali_Console](https://user-images.githubusercontent.com/1312165/82384606-13fb0580-99e5-11ea-98f1-0d9adeb51f93.png)

**Describe the solution you'd like**
Somehow to tie the various versions of a workload together visually.  I leave it up to Ux to determine the way.

**Comment by jshaughn** [comment]
@mtho11 Maybe you could expand on what you are looking for from the list view.  There is a visual grouping in the topology view.   In the list view you have the alphabetical sort that groups the names in order.  It's not clear to me what a user would be trying to accomplish to merit a change here.

**Comment by lucasponce** [comment]
> Somehow to tie the various versions of a workload together visually. I leave it up to Ux to determine the way.

Please, before to involve UX in any way, let that team leads review this.

> I would like to see a stronger/tighter more visually related grouping of workloads. Currently, there is nothing to show that they are related (by version).

This is not right, you have the Applications view to that.

Why do I need to group the workloads on this list ?
You can filter by label, but the concept of "application" which is the stronger view, belongs to the Application view.

My first view is that you are trying to replicate some concepts that are clear well known in other areas.

**Comment by mtho11** [comment]
The application's view is irrelevant here for Ux experience. The main concept is how we group different versions of the same workload in this screen. Show me that these are not different workloads (as the screen implies) but just different versions of the same workload.

**Comment by mtho11** [comment]
@jshaughn  you asked earlier to expand upon this. There is a grouping of workload versions. I believe the workload versions are related and should somehow visually be shown to be related like is already done in the app-version graph. 

The reason I marked this as `area/ux` is for feedback on how these versions should be visually related in a workload list.  I have ideas as to how this should be visually expressed but Ux is the final say and they have great ideas. Just to through ideas out there:

- Badges/icons to show similar versions
- The ideal solution might be some kind of Row Expander for hierarchical grid expansion to show a workload and its underlying versions

But these are all for Ux to decide...

**Comment by mtho11** [comment]
So we only want the Graph view to show versioning information as related/grouping right now?

**Comment by lucasponce** [comment]
> The application's view is irrelevant here for Ux experience. The main concept is how we group different versions of the same workload in this screen. Show me that these are not different workloads (as the screen implies) but just different versions of the same workload.

This is not correct @mtho11, "different versions of the same workload" is ambiguous.

> The reason I marked this as area/ux is for feedback on how these versions should be visually related in a workload list.

Please, let's stop to adding things to UX unless the feature is discussed and clear.
We have a limited bandwidth of UX resources and these kind of requests can move UX from the priorities of the next Sprints.

> I have ideas as to how this should be visually expressed but Ux is the final say and they have great ideas. Just to through ideas out there:

The Workload list is going to show basically that, there is a WIP to filter by label on a click, which is pretty much close what you are commenting.

But this creates a concern that we are not understanding here what an Application or a Workload is.

The Application is the "logical group" of workloads.

> So we only want the Graph view to show versioning information as related/grouping right now?

That's not correct, we have the Application/Workload/Services together.

**Comment by lucasponce** [comment]
Also, as a side note, basically this feature is also being implemented on this 
https://github.com/kiali/kiali/issues/2735

Where you can automatically filter by a given label clicking on it.

(Which is what you want here, to perform a filter per labels).

But the logical entity to group Workloads is the Application, this is why we have that concept in a separate list as similar as services.

**Comment by jshaughn** [comment]
> The ideal solution might be some kind of Row Expander for hierarchical grid expansion to show a workload and its underlying versions

@mtho11 OK thanks for adding detail to the RFE, I understand what you are looking for here, some sort of hierarchical view via row expansion or tree.   I'm not really opposed but I don't think the there is enough value to justify the work at this time.  I'm OK with leaving it open and seeing if we get community upvotes.

**Comment by lucasponce** [comment]
I don't think this one should be open.
The feature described is already implemented in two clear ways:

From the Workloads page using filtering:
![image](https://user-images.githubusercontent.com/1662329/97709158-2dd36500-1aba-11eb-86c2-09a787739556.png)

And conceptually we created the Application model for that:
![image](https://user-images.githubusercontent.com/1662329/97709234-480d4300-1aba-11eb-87ce-f6e7181183c8.png)
